GirafAPI/Endpoints/ActivityEndpoints.cs

```diff
@@ -301,49 +301,49 @@ await dbContext.Entry(citizen)
         
         // PUT updated activity
         group.MapPut("/activity/{id:int}", async (int id, UpdateActivityDTO updatedActivity, GirafDbContext dbContext) =>
-        {
-            try
             {
-                var activity = await dbContext.Activities.FindAsync(id);
-
-                if (activity is null)
+                try
                 {
-                    return Results.NotFound("Activity not found.");
-                }
+                    var activity = await dbContext.Activities
+                        .Include(a => a.Pictogram) 
+                        .FirstOrDefaultAsync(a => a.Id == id);
 
-                if (updatedActivity.PictogramId is not null)
-                {
+                    if (activity is null)
+                    {
+                        return Results.NotFound("Activity not found.");
+                    }
+                    
                     var pictogram = await dbContext.Pictograms.FindAsync(updatedActivity.PictogramId);
```

Comment:
if (updatedActivity.PictogramId is not null)
{
    var pictogram = await dbContext.Pictograms.FindAsync(updatedActivity.PictogramId);
    if (pictogram is null)
    {
        return Results.BadRequest($"Pictogram with ID {updatedActivity.PictogramId} not found.");
    }
    activity.Pictogram = pictogram;
}

Then update the individually updated properties like this(prevents updating unintended properties):
activity.Name = updatedActivity.Name;
activity.Description = updatedActivity.Description;
activity.Date = updatedActivity.Date;
activity.StartTime = updatedActivity.StartTime;
activity.EndTime = updatedActivity.EndTime;
activity.IsCompleted = updatedActivity.IsCompleted;

This avoids using SetValues and ToEntity that can unintentionally mess things up.
It also shouldn't have been used in the original

Reply:
The primary purpose of this pull request is to fix a bug that prevented pictograms from updating, even when the correct data was provided in the request. If you prefer a different structure that deviates from the original implementation, feel free to overwrite it and make the necessary updates. In that case, it might be more logical to apply the updates consistently throughout, rather than limiting them to this instance.